borg/archive.py

```diff
@@ -231,7 +233,7 @@ def add_item(self, item):
         if self.show_progress:
             self.stats.show_progress(item=item, dt=0.2)
         self.items_buffer.add(item)
-        if time.time() - self.last_checkpoint > self.checkpoint_interval:
+        if self.checkpoint_interval and time.time() - self.last_checkpoint > self.checkpoint_interval:
```

Comment:
rewrite uses checkpoint_interval=0 (because it handles that explicitly). Previously this would mean something like "write a checkpoint for every item" (with a small timing dependency?).